core/model/model.go

```diff
@@ -49,6 +42,40 @@ func (m ModelType) IsValid() bool {
 	return false
 }
 
+// Model represents the state of a model.
+type Model struct {
+	// Name returns the human friendly name of the model.
+	Name string
+
+	// UUID is the universally unique identifier of the model.
+	UUID UUID
+
+	// ModelType is the type of model.
+	ModelType ModelType
+
+	// AgentVersion is the target version for agents running under this model.
+	AgentVersion version.Number
+
+	// Cloud is the name of the cloud to associate with the model.
+	// Must not be empty for a valid struct.
+	Cloud string
+
+	// CloudRegion is the region that the model will use in the cloud.
+	CloudRegion string
+
+	// Credential is the id attributes for the credential to be associated with
+	// model. Credential must be for the same cloud as that of the model.
+	// Credential can be the zero value of the struct to not have a credential
+	// associated with the model.
+	Credential credential.ID
```

Comment:
Doesn't need to be changed as part of this PR. More of a question for you in general but I am thinking this needs to get swapped over to credential id (uuid) at some stage. Get us off this natural key as early as possible.